src/arr/compiler/anf-loop-compiler.arr

```diff
@@ -520,6 +520,37 @@ var total-time = 0
 
 show-stack-trace = false
 fun compile-fun-body(l :: Loc, step :: A.Name, fun-name :: A.Name, compiler, args :: List<N.ABind>, opt-arity :: Option<Number>, body :: N.AExpr, should-report-error-frame :: Boolean, is-flat :: Boolean, is-method :: Boolean) -> J.JBlock block:
+  var in-lam = false
+  var arg-used-in-lambda = false
+  arg-names = args.map(_.id)
+  dummy-anf-lettable = N.a-obj(A.dummy-loc, empty)
+  body.visit(N.default-map-visitor.{
```

Comment:
This would be better done with `default-iter-visitor`, so that you don't have to fake up a `dummy-anf-lettable` here.